### PR TITLE
Spike buffer realloc

### PR DESCRIPTION
**Pull request by fouriaux**
* use std::vector instead of new allocated memory, permitting easy realloc. 
* filter negative gid inside spike_buffer (shouldnt it be already filtered in setup anyway ?)

### CODE REVIEW

File: coreneuron/nrnoc/multicore.c [hunk 1]
```diff
@@ -444,22 +444,23 @@ void nrn_thread_table_check() {
 void nrn_multithread_job(void* (*job)(NrnThread*)) {
     int i;
 #if defined(_OPENMP)
-
+    printf("\n..openmp \n");
```
Comment (pramodk) [inline coreneuron/nrnoc/multicore.c]: these changes for `multicore.c` are coming from your local changes...

**Comment by fouriaux** [comment]
I have just reverted those changes on multicore.c

**Comment by fouriaux** [comment]
After validation, everything is working as expected.

**Comment by fouriaux** [comment]
I have just updated with changes required

**Pull request merged**